network/connPerfMon.go

```diff
@@ -298,31 +305,58 @@ func (pm *connectionPerformanceMonitor) updateMessageIdlingInterval(now int64) (
 
 func (pm *connectionPerformanceMonitor) pruneOldMessages(now int64) {
 	oldestMessage := now - int64(pmMaxMessageWaitTime)
-	for digest, pendingMsg := range pm.pendingMessages {
-		if oldestMessage < pendingMsg.firstPeerTime {
+	prunedBucketsCount := 0
+	for bucketIdx, currentMsgBucket := range pm.pendingMessagesBuckets {
+		if currentMsgBucket.endTime > oldestMessage {
+			pm.pendingMessagesBuckets[bucketIdx-prunedBucketsCount] = currentMsgBucket
```

Comment:
what is the point? if `prunedBucketsCount == 0` and `bucketIdx == 0` then assigning self. plus, `prunedBucketsCount` and `bucketIdx` and synchronized and incremented on each iteration, so the index is always zero?

Reply:
no, the `prunedBucketsCount` is not always identical to the `bucketIdx`. That's why I added a `continue`.
( I created a unit test `TestBucketsPruning` to verify that )

Reply:
ah, right, my bad